### ISSUE TITLE
Getting an Error when mining Cryptonight

### ISSUE THREAD

**Issue by izzledude** [MultiPoolMiner/MultiPoolMiner#1562]
2018-04-08 16:41:54 VERBOSE: XmrigNvidia: [CUDA] Error gpu 0: <cryptonight_extra_cpu_final>:232 "the launch timed out an
d was terminated"
2018-04-08 16:41:54 INFO: Starting next run.
2018-04-08 16:41:55 INFO: Updating exchange rates from Coinbase.
2018-04-08 16:41:55 INFO: Loading saved statistics.
2018-04-08 16:41:58 INFO: Loading pool information.
2018-04-08 16:42:05 INFO: Selecting best pool for each algorithm.
2018-04-08 16:42:06 INFO: Getting miner information.
2018-04-08 16:42:09 INFO: Calculating profit for each miner.
2018-04-08 16:42:11 INFO: Starting miner (FireiceCpu): 'Z:\Desktop\MPM 2.7.2.7\Bin\CryptoNight-FireIce\xmr-stak.exe -c M
iningPoolHub_CryptoNight_IzzleDude.GPC_Cpu.txt --noUAC --noAMD --noNVIDIA'
2018-04-08 16:42:11 INFO: Starting miner (XmrigNvidia): 'Z:\Desktop\MPM 2.7.2.7\Bin\CryptoNight-NVIDIA\xmrig-nvidia.exe
--api-port 3335 -a cryptonight -o stratum+tcp://us-east.cryptonight-hub.miningpoolhub.com:17024 -u IzzleDude.GPC -p x --
keepalive --nicehash --donate-level 1'
2018-04-08 16:42:11 INFO: Pinging monitoring server.
2018-04-08 16:42:11 INFO: Miner Status (https://multipoolminer.io/monitor/miner.php): Success
2018-04-08 16:42:11 INFO: Start waiting before next run.
2018-04-08 16:43:01 INFO: Finish waiting before next run.
2018-04-08 16:43:01 INFO: Saving hash rates.
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * VERSIONS:     XMRig/2.4.5 libuv/1.18.0 CUDA/9.0 MSVC/2015
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * CPU:          AMD Ryzen 7 1800X Eight-Core Processor          (1) x64 AES-N
I
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * GPU #0:       GeForce GTX 1070 @ 1835/4004 MHz 64x45 6x25 arch:61 SMX:15
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * ALGO:         cryptonight, donate=1%
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * POOL #1:      us-east.cryptonight-hub.miningpoolhub.com:17024
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * API PORT:     3335
2018-04-08 16:43:03 VERBOSE: XmrigNvidia:  * COMMANDS:     'h' hashrate, 'e' health, 'p' pause, 'r' resume
2018-04-08 16:43:03 VERBOSE: XmrigNvidia: [2018-04-08 16:42:13] use pool us-east.cryptonight-hub.miningpoolhub.com:17024
 172.104.252.176
2018-04-08 16:43:03 VERBOSE: XmrigNvidia: [2018-04-08 16:42:13] new job from us-east.cryptonight-hub.miningpoolhub.com:1
7024 diff 500054
2018-04-08 16:43:03 VERBOSE: XmrigNvidia: [2018-04-08 16:42:13] new job from us-east.cryptonight-hub.miningpoolhub.com:1
7024 diff 40000
2018-04-08 16:43:03 VERBOSE: XmrigNvidia: [2018-04-08 16:42:21] new job from us-east.cryptonight-hub.miningpoolhub.com:1
7024 diff 40000
2018-04-08 16:43:03 VERBOSE: XmrigNvidia: [CUDA] Error gpu 0: <cryptonight_extra_cpu_final>:232 "the launch timed out an
d was terminated"

**Comment by 3MUl0R** [comment]
This is probably due to the monero fork.  All of the monero miners need to be updated.

**Comment by paulpoco** [comment]
Is there config to change?

**Comment by UselessGuru** [comment]
> Is there config to change?

No, not for XmrigNvidia. Just download and replace the binaries (this is to the best of my knowledge, I cannot test because I don't own AMD hardware).
Just try - you cannot break anything. :)

**Comment by izzledude** [comment]
@UselessGuru Ok i did those changes of the binaries and will try and report back. if i get the issue again whats the next step?

@titvs  where would/how would i add those parameters?

**Comment by titvs** [comment]
@izzledude You need to edit the miner file **XmrigNvidia.txt**, and add the "--config=config.json" parameter. Like this:

`"Arguments":  "\"--api-port 3335 --config=config.json -a cryptonight -o $($Pools.CryptoNight.Protocol)://$($Pools.CryptoNight.Host):$($Pools.CryptoNight.Port) -u $($Pools.CryptoNight.User) -p $($Pools.CryptoNight.Pass) --keepalive --nicehash --donate-level 1\"",`

After MPM tries to benchmark the miner and it fails, edit the config.json file and change the threads value. I advise you to run the miner manually until you reach a threads value that's stable for your system.

**Comment by izzledude** [comment]
Looking at the main page on MPH it shows the monero algo was changed  do we need to change that on the start files also?

**Comment by 3MUl0R** [comment]
The example shown on the MPH monero pool page is specifically for the ClaymoreAmd miner.  afaik if you want to use this miner you need to update the miner binary (found in the bin folder) and then update the launch string with the -pow7 1 option.

I manually added the **-pow7 1** to line 5 of Miners\ClaymoreAmd.txt

>  "Arguments":  "\"-r -1 -mport -13333 -pow7 1 -xpool $($Pools.CryptoNight.Protocol)://$($Pools.CryptoNight.Host):$($Pools.CryptoNight.Port) -xwal $($Pools.CryptoNight.User) -xpsw $($Pools.CryptoNight.Pass)\"",

**Comment by izzledude** [comment]
So semi related question so if a miner gives the error "Unable to update"  or "Update failed"   does that mean its a situation like @UselessGuru  mentioned above that i need to manually update the binaries ?

**Comment by izzledude** [comment]
Also is there a "noob" forum where i can ask more basic type questions. I'd hate to open cases for basic questions when everyone here is dealing with more important stuff.

**Comment by 3MUl0R** [comment]
You could try reddit: https://www.reddit.com/r/multipoolminer/

**Comment by UselessGuru** [comment]
> You could try reddit: https://www.reddit.com/r/multipoolminer/

You'd probably see the same people keeping an eye on it. :)

**Comment by UselessGuru** [comment]
@titvs 

"--config=config.json" should not be needed unless you want to do some extra tweaking. We would like to keep things as simple as possible here. :-)

**Comment by titvs** [comment]
@UselessGuru I know but xmrig-nvidia is a very sensible beast and there are a lot of people with this kind of problem that can be solved by adjusting the threads, because the miner crashes with the default values.
You can only adjust them if you include the config.json switch because it only writes (and later reads) the default values to this file when this parameter is enabled.

**Comment by izzledude** [comment]
@UselessGuru  I posted on that forum over two weeks ago but unfortunately had no luck its why i thought about asking if there was any other place for help. 

Here is the post if anyone has any insight 
https://www.reddit.com/r/multipoolminer/comments/878vyx/questions_about_current_set_up_need_advice/

**Comment by izzledude** [comment]
As for the error I have not seen it since dong the change that @UselessGuru  suggested. I haven't tried @titvs  suggestions just because i dont have much experience with MPM so i dont want to mess anything up and then have to re download everything again .

**Comment by titvs** [comment]
@izzledude it’s possible that your system now runs ok with the suggested defaults from the new version miner. If it’s running normally you don’t need to change anything. ;)
